src/lexedata/importer/fromexcel.py

```diff
@@ -576,19 +573,22 @@ def load_dataset(
     EP.db.empty_cache()
     EP.parse_cells(lexicon_wb)
     # load cognate data set if provided by metadata
-    cognate = True if cognate_lexicon and dialect.cognates else False
-    if cognate:
-        try:
+    try:
+        dialect = argparse.Namespace(**dataset.tablegroup.common_props["special:fromexcel"])
+        cognate = True if cognate_lexicon and dialect.cognates else False
+        if cognate:
             ECP = excel_parser_from_dialect(
                 argparse.Namespace(**dialect.cognates), cognate=cognate
             )
-        except KeyError:
-            ECP = ExcelCognateParser(dataset, db_fname=db)
-        ECP = ECP(dataset, db)
-        ECP.db = EP.db
-        for sheet in openpyxl.load_workbook(cognate_lexicon).worksheets:
-            EP.parse_cells(sheet)
-    EP.db.write_dataset_from_cache()
+        else:
+            raise KeyError
+    except KeyError:
```

Comment:
This shouldn't be de-dented. Sometimes, we might want to import a lexical dataset without cognates.